lib/clean-git.nix

```diff
@@ -24,6 +24,14 @@ with builtins;
 # is shared among multiple invocations of gitSource:
 
 let
+  # This avoids accessing src unfiltered when testing for the existance
+  # and type of these paths.  Important for `lorri` support.
+  # See https://github.com/input-output-hk/haskell.nix/issues/492
+  justDotGitAndIndex = cleanSourceWith {
```

Comment:
Have you tried this? I'm concerned this will still end up with lorri watching the whole git dir. I'll give it a shot and see.

Reply:
Seems to work, nice!

Reply:
I'm a bit mystified as to *why* it works, but hey :sweat_smile:

Reply:
In particular, I added a new file and committed, and it didn't trigger a re-eval, which actually seems kind of suspicious.